src/preact/base-element.js

```diff
@@ -238,20 +247,43 @@ export class PreactBaseElement extends AMP.BaseElement {
             'when configured with "children", "passthrough", or ' +
             '"passthroughNonEmpty" properties.'
         );
-        const shadowRoot = this.element.attachShadow({mode: 'open'});
-        this.container_ = shadowRoot;
-
-        const shadowCss = Ctor['shadowCss'];
-        if (shadowCss) {
-          installShadowStyle(shadowRoot, this.element.tagName, shadowCss);
+        // Check if there's a pre-constructed shadow DOM.
+        let {shadowRoot} = this.element;
+        let container = shadowRoot && childElementByTag(shadowRoot, 'c');
+        if (container) {
+          toHydrate = true;
```

Comment:
I think this is going to cause every rerender to be a hydrate. Is that right?

Reply:
This part of code executes only on first render. All other renders bypass this branch.

Reply:
I found an easy way to test it too. Added tests. PTAL.